lib/gui/models/flash-state.js

```diff
@@ -132,8 +133,8 @@ FlashState.service('FlashStateModel', function() {
         speed: _.attempt(() => {
           if (_.isNumber(state.speed) && !_.isNaN(state.speed)) {
 
-            // Transform bytes to megabytes preserving only two decimal places
-            return Math.floor(state.speed / 1e+6 * 100) / 100;
+            // Preserve only two decimal places
+            return Math.floor(units.bytesToMegabytes(state.speed) * 100) / 100;
```

Comment:
IMHO it's always best to keep maximum-precision internally (i.e. keep the internal speed as bytes/sec) and then only trim precision and round to 'human-friendly' units at the last stage possible.